src/ui/map.js

```diff
@@ -599,6 +608,49 @@ class Map extends Camera {
     getMaxZoom() { return this.transform.maxZoom; }
 
     /**
+     * @private
+     */
+    _transformRequest(url: string, resourceType?: ResourceTypeEnum): RequestParameters {
+        let requestParameters: RequestParameters = {
+            url: url,
+            headers: {}
+        };
+
+        if (!this._transformRequestCallback) {
+            return requestParameters;
+        }
+
+        requestParameters = this._transformRequestCallback(url, resourceType || ajax.ResourceType.Unknown) || requestParameters;
+        return requestParameters;
+    }
+
+    /**
+     * Sets the map's transform request callback function.
+     * The callback can be used to modify the url, set headers, or set the withCredentials property for cross-origin requests.
+     * 
+     * @function setRequestTransform
+     * @param {Function} callback A callback run before the Map makes a request for an external URL. Called with a url and resourceType.
+     *  Expected to return an object with a `url` property and optionally `headers` and `withCredentials` properties.
+     * 
+     * If called with no arguments, the existing callback will be removed.
+     * 
+     * @example
+     * Map.setRequestTransform( (url, resourceType)=> {
+     *  if(resourceType == 'Source' && url.startsWith('http://myHost') {
+     *   return {
+     *    url: url.replace('http', 'https'),
+     *    headers: { 'my-custom-header': true},
+     *    withCredentials: true
+     *  }
+     * });
+     */
+    setRequestTransform(transform: Function) {
```

Comment:
Thanks for this suggestion. I didn't even consider the timing and order-of-execution issue with regards to `Map#setRequestTransform.` 
`reousrceType` is optional for catching calls that don't provide a type.